src/analysis/calculus/taylor.lean

```diff
@@ -358,16 +358,15 @@ begin
   have h' : ∀ (y : ℝ) (hy : y ∈ Ico a x),
     ∥((n! : ℝ)⁻¹ * (x - y) ^ n) • iterated_deriv_within (n + 1) f (Icc a b) y∥
     ≤ (n! : ℝ)⁻¹ * |(x - a)|^n * C,
-  { intros y hy,
+  { rintro y ⟨hay, hyx⟩,
     rw [norm_smul, real.norm_eq_abs],
     -- Estimate the iterated derivative by `C`
-    refine mul_le_mul _ (hC y ⟨hy.1, hy.2.le.trans hx.2⟩) (by positivity) (by positivity),
+    refine mul_le_mul _ (hC y ⟨hay, hyx.le.trans hx.2⟩) (by positivity) (by positivity),
     -- The rest is a trivial calculation
     rw [abs_mul, abs_pow, abs_inv, nat.abs_cast],
-    mono*,
+    mono* with [0 ≤ (n! : ℝ)⁻¹],
```

Comment:
I'm still slightly worried by this, it seems like this PR has made `mono` weaker than it was before. Can you justify why this use of `mono` has been modified?

Reply:
Needing to add a side condition is actually a mark that `mono` got *stronger*: as it now has more applicable lemmas, we need to disambiguate.

Reply:
Not really - it means `mono` is harder to use, the invocation `mono` can do less than it could before. Just having more lemmas doesn't constitute an improvement (eg, `simp` wouldn't be better if every single lemma was marked simp!) In addition, do we have examples of `mono` doing more than it previously could, to justify the "stronger"?

Reply:
Indicating side goals seems to be a usual way of using `mono`. But I am no `mono` expert.